bin/wordle.raku

```diff
@@ -2,7 +2,11 @@
 
 use Games::Wordle;
 
-my Games::Wordle $wordle.=new;
+subset Number of Str where !.defined || (.Int.so && .Int > 0);
```

Comment:
I explained it in the commit message:

> Had to create a subset, because a lone `--number` would also be
> accepted, because Bool is Int, so `--number 1` wouldn't work, only
> `--number=1` would.
>
>Now `--number` is not accepted, but `--number 1` and `--number=1` are.

Because `Bool` is an `Int`. Specifying `--number` without a number will work and `$number` will contain `True`. And `--number 1` wouldn't work because `--number` can be used in two ways, so you have to write `--number=1`.
And `Int:D` has the same issue, plus it makes it a mandatory option.

Another option is providing a default value inside the `MAIN` sub(checking for Bool).

Another option is using the Getopt::Long module and simply writing `Int :$number`, but it may be overkill depending on what you plan to do.

Reply:
Ah I see, I wasn't aware of `--number` being picked up as a Bool. I think it could probably be simplified with using `IntStr :$number` and checking `.chars`, but I might just do that in a later commit. I'll probably make some changes to the lib in a future commit anyway so that it corrects the given number to be in range of how many answers there are.